nevergrad/optimization/optimizerlib.py

```diff
@@ -485,28 +487,25 @@ def _internal_tell(self, x: base.ArrayLike, value: float) -> None:
                 self.llambda = max(self.llambda, self.num_workers)
                 self.mu = self.llambda // 4
             self.archive_fitness = []
-        idx = self.unevaluated_population.index(tuple(x))
-        self.evaluated_population += [x]
-        self.evaluated_population_fitness += [value]
-        self.evaluated_population_sigma += [self.unevaluated_population_sigma[idx]]
-        del self.unevaluated_population[idx]
-        del self.unevaluated_population_sigma[idx]
+        x = np.array(x, copy=False)
+        x_bytes = x.tobytes()
+        particule = self.unevaluated_population[x_bytes]
+        particule.fitness = value
+        self.evaluated_population.append(particule)
         if len(self.evaluated_population) >= self.llambda:
             # Sorting the population.
-            sorted_pop_with_sigma_and_fitness = [(i, s, f) for f, i, s in sorted(
-                zip(self.evaluated_population_fitness, self.evaluated_population, self.evaluated_population_sigma))]
-            self.evaluated_population = [p[0] for p in sorted_pop_with_sigma_and_fitness]
-            self.evaluated_population_sigma = [p[1] for p in sorted_pop_with_sigma_and_fitness]
-            self.evaluated_population_fitness = [p[2] for p in sorted_pop_with_sigma_and_fitness]
+            self.evaluated_population.sort(key=lambda p: p.fitness)
             # Computing the new parent.
-            self.current_center = sum([np.asarray(self.evaluated_population[i]) for i in range(self.mu)]) / self.mu
-            self.sigma = np.exp(sum([np.log(self.evaluated_population_sigma[i]) for i in range(self.mu)]) / self.mu)
+            self.current_center = sum(p.position for p in self.evaluated_population[:self.mu]) / self.mu
+            self.sigma = np.exp(np.sum(np.log([p.sigma for p in self.evaluated_population[:self.mu]])) / self.mu)
             self.evaluated_population = []
-            self.evaluated_population_sigma = []
-            self.evaluated_population_fitness = []
+        del self.unevaluated_population[x_bytes]
 
     def tell_not_asked(self, x: base.ArrayLike, value: float) -> None:
-        raise base.TellNotAskedNotSupportedError
+        x = np.array(x, copy=False)
+        self.unevaluated_population[x.tobytes()] = ParticuleTBPSA(x, sigma=self.sigma)
```

Comment:
@teytaud 
For inoculation, what should be sigma ? Should it depend on how far the point is from the center?